tests/e2e-pw/tests/cart-checkout/cart-block.merchant.block_theme.spec.ts

```diff
@@ -0,0 +1,194 @@
+			);
+
+			const addBlockButton = await editor.canvas
+				.getByRole( 'document', { name: 'Block: Order Summary' } )
+				.getByRole( 'button', { name: 'Add block' } );
+			await addBlockButton.dispatchEvent( 'click' );
+			await editor.page
+				.getByLabel( 'Search for blocks and patterns' )
+				.fill( 'Table' );
+			const tableButton = await editor.page.getByRole( 'option', {
+				name: 'Table',
+			} );
+			await expect( tableButton ).toBeVisible();
+
+			await editor.page
+				.getByLabel( 'Search for blocks and patterns' )
+				.fill( 'Audio' );
+
+			const audioButton = await editor.page.getByRole( 'option', {
+				name: 'Audio',
+			} );
+			await test.expect( audioButton ).toBeVisible();
+
+			// Now check the filled cart block and expect only the Table block to be available there.
+			await editor.selectBlocks(
+				blockSelectorInEditor +
+					' [data-type="woocommerce/filled-cart-block"]'
+			);
+			const filledCartAddBlockButton = await editor.canvas
+				.getByRole( 'document', { name: 'Block: Filled Cart' } )
+				.getByRole( 'button', { name: 'Add block' } )
+				.first();
+			await filledCartAddBlockButton.click();
+
+			const filledCartTableButton = await editor.page.getByRole(
+				'option',
+				{
+					name: 'Table',
+				}
+			);
+			await expect( filledCartTableButton ).toBeVisible();
+
+			const filledCartAudioButton = await editor.page.getByRole(
+				'option',
+				{
+					name: 'Audio',
+				}
+			);
+			await expect( filledCartAudioButton ).not.toBeVisible();
+		} );
+
+		test( 'shows empty cart when changing the view', async ( {
+			page,
+			editor,
+			editorUtils,
+		} ) => {
+			await editor.selectBlocks( blockSelectorInEditor );
+			await editor.page
+				.getByRole( 'button', { name: 'Switch view' } )
+				.click();
+			const emptyCartButton = await page.getByRole( 'menuitem', {
+				name: 'Empty Cart',
+			} );
+
+			// Focus the empty cart button and wait for the tooltip to disappear.
+			await emptyCartButton.focus();
+			await page.waitForTimeout( 1000 );
+			await emptyCartButton.click();
+
+			const filledCartBlock = await editorUtils.getBlockByName(
+				'woocommerce/filled-cart-block'
+			);
+			const emptyCartBlock = await editorUtils.getBlockByName(
+				'woocommerce/empty-cart-block'
+			);
+			await expect( filledCartBlock ).not.toBeVisible();
+			await expect( emptyCartBlock ).toBeVisible();
+			await editor.selectBlocks( blockSelectorInEditor );
+			await page.getByRole( 'button', { name: 'Switch view' } ).click();
+
+			const filledCartButton = await page.getByRole( 'menuitem', {
+				name: 'Filled Cart',
+			} );
+
+			await filledCartButton.click();
+			await expect(
+				editor.canvas.locator(
+					blockData.selectors.editor.block +
+						' [data-type="woocommerce/filled-cart-block"]'
+				)
+			).toBeVisible();
+			await expect(
+				editor.canvas.locator(
+					blockData.selectors.editor.block +
+						' [data-type="woocommerce/empty-cart-block"]'
+				)
+			).not.toBeVisible();
```

Comment:
We can refactor this to `.toBeHidden`. I'm adding `eslint-plugin-playwright` package that has [this rule](https://github.com/playwright-community/eslint-plugin-playwright/blob/main/docs/rules/no-useless-not.md).

```suggestion
			).toBeHidden();
```